test/utils/trades.ts

```diff
@@ -73,3 +73,56 @@ export const toMinBuyAmt = (
 
   return divCeil(divCeil(product, highBuyPrice), fp('1')) // (c)
 }
+
+// Returns the buy amount in the auction for the given progression
+export const dutchBuyAmount = async (
+  progression: BigNumber,
+  assetInAddr: string,
+  assetOutAddr: string,
+  outAmount: BigNumber,
+  minTradeVolume: BigNumber,
+  maxTradeSlippage: BigNumber
+): Promise<BigNumber> => {
+  const assetIn = await ethers.getContractAt('IAsset', assetInAddr)
+  const assetOut = await ethers.getContractAt('IAsset', assetOutAddr)
+  const [sellLow, sellHigh] = await assetOut.price() // {UoA/sellTok}
+  const [buyLow, buyHigh] = await assetIn.price() // {UoA/buyTok}
+
+  const inMaxTradeVolume = await assetIn.maxTradeVolume()
+  let maxTradeVolume = await assetOut.maxTradeVolume()
+  if (inMaxTradeVolume.lt(maxTradeVolume)) maxTradeVolume = inMaxTradeVolume
+
+  const auctionVolume = outAmount.mul(sellHigh).div(fp('1'))
+  const slippage = maxTradeSlippage
+    .mul(
+      fp('1').sub(
+        auctionVolume.sub(minTradeVolume).mul(fp('1')).div(maxTradeVolume.sub(minTradeVolume))
+      )
+    )
+    .div(fp('1'))
+
+  // console.log('slippage: ', slippage)
+
+  const lowPrice = sellLow.mul(fp('1').sub(slippage)).div(buyHigh)
+  const middlePrice = divCeil(sellHigh.mul(fp('1')), buyLow)
+  const highPrice = middlePrice.add(divCeil(middlePrice, bn('2'))) // 50% above middlePrice
+
+  const price = progression.lt(fp('0.15'))
+    ? highPrice.sub(highPrice.sub(middlePrice).mul(progression).div(fp('0.15')))
+    : middlePrice.sub(
+        middlePrice
+          .sub(lowPrice)
+          .mul(progression.sub(fp('0.15')))
+          .div(fp('0.85'))
+      )
+  // console.log(
```

Comment:
delete

Reply:
really useful for debugging! but okay, yeah, deleting